### ISSUE TITLE
extension breaks after changing a test file

### ISSUE THREAD

**Issue by jd1378** [vitest-dev/vscode#393]
### Describe the bug

In Testing panel of vscode vitest throws `AssertionError [ERR_ASSERTION]: root directory is required to serve files` after saving a file.
extension works correctly when vscode is opened for the first time and nothing is changed in files yet.

I wish I could create a reproduction, but I can't

the only thing I'm noticing is in the logs it's missing the last letter of the path: `test/client.test.t`

### Reproduction

-

### Output

```shell
[Worker] Collecting tests due to file changes: test/client.test.t
[INFO 9:08:19 AM] [TEST] WARNING: NODE_ENV value of 'test' did not match any deployment config file names.

[INFO 9:08:19 AM] [TEST] WARNING: See https://github.com/node-config/node-config/wiki/Strict-Mode

[INFO 9:08:19 AM] [TEST] WARNING: No configurations found in configuration directory:C:\Users\MY User\AppData\Local\Programs\Microsoft VS Code\config

[INFO 9:08:19 AM] [TEST] WARNING: To disable this warning set SUPPRESS_NO_CONFIG_WARNING in the environment.
```

### Version

v0.10.7

### Validations

- [X] Check that you are using the latest version of the extension
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vscode/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vscode/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by jd1378** [comment]
Well It seems the error is happening in my source code somewhere when it's running `app.use(serveStatic(app.get('public')))`,
which seems to get some config, and that config is not loaded

it seems the working directory of the extension is not at the project, but somewhere in vscode folder in appdata

what can I do to set working directory to my project ?

**Comment by hornta** [comment]
I'm getting the same problem. This happens because you are rerunning the tests too quickly after changing the test file. You'll have to wait a bit, 5 seconds or so, after changing the file.

**Comment by jd1378** [comment]
This might be windows only, but the problem is not frequencbeing too quickly, it outright crashes, because the working directory is set to somewhere other than the project

**Comment by hornta** [comment]
Yes I've also observed the cwd being incorrect which caused my test suite to fail. I could bypass that issue by finding the correct project folder by traversing backwards from `__dirname` but it's not elegant. For me, when I rerun the tests quickly after changing the test file, the test looks like they are running because a small loader appears in the top of the test explorer pane, but no test actually runs I think, because no console logs are recorded. I've never had vscode outright crash on me though.

**Comment by gerhardcit** [comment]
I would suggest this is not a minor bug.
The same thing happens while running pnpm dev in sveltekit project and when hmr reloads a file etc:
`src/routes/api/someroute/[id]/+page.server.ts`

The extension crashes with an error saying: 'src/routes/api/someroute/[id]/+page.server.ts' is not available.
Then it just hangs. No way of resetting or restarting. The only way is to close vscode and reopen.
It's very annoying, and basically makes the extension unusable. So hence my suggestion this is NOT a minor bug.

You end up being stuck with spinner and "Resolving Vitest" and no way to "hard reset" or do something. Surely that must be an option?

**Comment by sheremet-va** [comment]
I need some kind of reproduction that I can install and run to see the error by myself.

**Comment by sheremet-va** [comment]
> Then it just hangs. No way of resetting or restarting. The only way is to close vscode and reopen.

There is a "Refresh" button that should work:

![Screenshot 2024-06-05 at 10 32 48](https://github.com/vitest-dev/vscode/assets/16173870/d966ca99-e1c1-406c-91fd-ca34734d4964)

If it doesn't I will need a reproduction to see why.

**Comment by sheremet-va** [comment]
The potential fix is released in v0.11.0 - this is a pre-release version and you might need to enable it manually:

![Screenshot 2024-06-05 at 10 48 25](https://github.com/vitest-dev/vscode/assets/16173870/f18c2317-07a2-47dd-b033-1067ed772b46)

Please let me know if it fixes your issue.

**Comment by jd1378** [comment]
thanks a lot for the quick fix
it seems to be working for me

**Comment by sheremet-va** [comment]
This should be fixed for everyone in 0.12.0.
